cmd/kubeadm/README.md

```diff
@@ -0,0 +1,97 @@
+# Kubernetes Cluster Boostrap Made Easy
+
+## Usage
+
+### `kubeadm init`
+
+It's usually enough to run `kubeadm init`, but in some case you might like to override the
+default behaviour. The flags used for said purpose are described below.
+
+- `--token=<token>`
+
+By default, a token is generated, but if you are to automate cluster deployment, you will want to
+set the token ahead of time. Read the docs for more information on the token format.
+
+- `--api-advertise-addresses=<ips>` (multiple values are allowed by having multiple flag declarations or multiple values separated by comma)
+- `--api-external-dns-names=<domain>` (multiple values are allowed by having multiple flag declarations or multiple values separated by comma)
+
+By default, `kubeadm` will auto detect IP addresses and use that to generate API server certificates.
+If you would like to access the API via any external IPs and/or hostnames, which it might not be able
+to detect, you can use `--api-advertise-addresses` and `--api-external-dns-names` to add multiple
+different IP addresses and hostnames (DNS).
+
+- `--service-cidr=<cidr>` (default: "100.64.0.0/12")
+
+By default, `kubeadm` sets `100.64.0.0/12` as the subnet for services. This means when a service is created, its cluster IP, if not manually specified, 
+will be automatically assigned from the services subnet. If you would like to set a different one, use `--service-cidr`.
+
+- `--service-dns-domain=<domain>` (default: "cluster.local")
+
+By default, `kubeadm` sets `cluster.local` as the cluster DNS domain. If you would like to set a different one, use `--service-dns-domain`.
+
+- `--schedule-workload=<bool>` (default: "false")
+
```

Comment:
Good question, I am not entierely sure why we ended-up with this exact flag. @lukemarsden might have some background on this.